kong/plugins/zipkin/opentracing.lua

```diff
@@ -7,15 +7,15 @@ A plugin that derives this should:
   - Implement a :initialise_request(conf, ctx) method if it needs to do per-request initialisation
 ]]
 
-local BasePlugin = require "kong.plugins.base_plugin"
 local subsystem = ngx.config.subsystem
 
-local OpenTracingHandler = BasePlugin:extend()
-OpenTracingHandler.VERSION = "scm"
+local OpenTracingHandler = {
+  VERSION = "scm",
+  -- We want to run first so that timestamps taken are at start of the phase
+  -- also so that other plugins might be able to use our structures
+  PRIORITY = 100000,
+}
 
--- We want to run first so that timestamps taken are at start of the phase
--- also so that other plugins might be able to use our structures
-OpenTracingHandler.PRIORITY = 100000
 
 function OpenTracingHandler:new(name)
```

Comment:
Is this `:new` function still being used with the removal of BasePlugin?

In other plugins were BasePlugin was removed, the `:new` operation was removed altogether. Generally, it was just boilerplate, but in the case of this plugin it does a thing in line 23 below which I'm not seeing in Kong's plugin handling how it ever gets executed (though I may be just missing it).

Reply:
Agreed, I missed the `new` function. Removing it.